File: cs/sample/Sample/ListNode/ListNode.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Sample.ListNode{
    public class listNode{
        public int val;
        public listNode next;
        public listNode(int val=0,listNode next=null){
            this.val = val;
            this.next = next;
        }
        public static string strForm(int val,string s,string mode="reverse"){
            if (mode=="reverse") return val.ToString()+s;
            else return s+val.ToString();
        }
        public static string printListNode(listNode l,string mode="reverse"){
            string printStr = "";
            while (l!=null){
                int tmp = l.val;
                printStr = listNode.strForm(tmp,printStr,mode);
                l = l.next;
            }
            return printStr;
        }
        public static listNode listNode_fromStr(string s){
            listNode result = new listNode();
            listNode tmp = result;
            foreach (char c in s){
                int tmpint = (int)Char.GetNumericValue(c);
                tmp.next = new listNode(tmpint);
                tmp = tmp.next;
            }
            return result.next;
        }
        public static listNode listNode_fromStrReverse(string s){
            int i=0;
            listNode result = new listNode((int)Char.GetNumericValue(s[0]));
            listNode prev = result;
            foreach (char c in s.Skip(1)){
                int tmpint = (int)Char.GetNumericValue(c);
                result = new listNode(tmpint,prev);
                prev = result;
                i+=1;
            }
            return result;
        }
    }
}

File: cpp/example/src/ListNode/listNode.cpp
#include "listNode.h"

string ListNode::strForm(int val,string s,string mode){
    if (mode=="reverse")
        return to_string(val)+s;
    else return s+to_string(val);
}

string* ListNode::printListNode(ListNode* l,string mode){
    string* printStr = new string("");
    while (l != nullptr){
        int tmp = l->val;
        *printStr = ListNode::strForm(tmp,*printStr,mode);
        l = l->next;
    }
    return printStr;
}

ListNode* ListNode::listNode_fromStr(string s){
    ListNode* result = new ListNode();
    ListNode** tmp = &result;
    for (char c:s){
        int tmpint = c-'0';
        (*tmp)->next = new ListNode(tmpint);
        tmp = &((*tmp)->next);
    }
    return result->next;
}

ListNode* ListNode::listNode_fromStrReverse(string s){
    int i=0;
    ListNode* result;
    ListNode* prev;
    for (char c:s){
        int tmp = c-'0';
        if (i==0) {result = new ListNode(tmp);}
        else {result = new ListNode(tmp,prev);}
        prev = result;
        i+=1;
    }
    return result;
}